### PR TITLE
Added more contexts to errors in queue and downloader

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- This PR is derived from ethereum/go-ethereum#21067
- This PR should be merged after #788

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jeongkyun-oh** [comment]
I will open this PR after #788 merged.

**Review by ehnuje — DISMISSED** [review]

File: datasync/downloader/downloader.go [hunk 1]
```diff
@@ -316,12 +316,20 @@ func (d *Downloader) UnregisterPeer(id string) error {
...
-	case errBusy, errCanceled:
+	case nil, errBusy, errCanceled:
+		return err
+	}
+
+	if errors.Is(err, errInvalidChain) {
+		logger.Warn("Synchronisation failed, dropping peer", "peer", id, "err", err)
+		d.dropPeer(id)
+		return err
+	}
```
Comment (aidan-kwon) [inline datasync/downloader/downloader.go:327]: Why did you omit the `nil` handling for `d.dropPeer`?

Comment (ehnuje) [reply datasync/downloader/downloader.go:327]: Maybe because of the missing command `copydb` in Klaytn?

Comment (KimKyungup) [reply datasync/downloader/downloader.go:327]: I think even if we don't have that case, `nil` case check is useful.
@jeongkyun-oh Let's add it.

**Review by KimKyungup — COMMENTED** [review]
LGTM without the comment.

File: datasync/downloader/downloader.go (hunk 1 above)
Comment (jeongkyun-oh) [reply datasync/downloader/downloader.go:327]: @aidan-kwon Yes, `d.dropPeer` is nil only if `--copydb` flag is on. There is no `nil` checking for the same logic already.
@KimKyungup I will apply them to other places as well if necessary for consistency on another PR.

**Review by aidan-kwon — DISMISSED** [review]

File: datasync/downloader/downloader.go (hunk 1 above)
Comment (KimKyungup) [reply datasync/downloader/downloader.go:327]: Yes. Please apply them for consistency.

**Review by KimKyungup — DISMISSED** [review]

**Comment by jeongkyun-oh** [comment]
@ehnuje PTAL.

**Review by ehnuje — APPROVED** [review]

**Review by ehnuje — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Review by KimKyungup — APPROVED** [review]

**Pull request merged**